File: specification/specs2/cluster/nodes_stats/NodesStatsRequest.ts
class NodesStatsRequest extends RequestBase {
	@request_parameter()
	completion_fields: Field[];
	@request_parameter()
	fielddata_fields: Field[];
	@request_parameter()
	fields: Field[];
	@request_parameter()
	groups: boolean;
	@request_parameter()
	level: Level;
	@request_parameter()
	types: string[];
	@request_parameter()
	timeout: Time;
	@request_parameter()
	include_segment_file_sizes: boolean;
}

File: output/csharp/Cluster/NodesStats/NodesStatsRequest.cs
using Nest.CommonAbstractions;
using Nest.Internal;
using Nest.Common;
using Nest.CommonOptions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cluster {

	public class NodesStatsRequest : RequestBase {
		
		[DataMember(Name="completion_fields")]
		public List<Field> CompletionFields { get; set; }


		[DataMember(Name="fielddata_fields")]
		public List<Field> FielddataFields { get; set; }


		[DataMember(Name="fields")]
		public List<Field> Fields { get; set; }


		[DataMember(Name="groups")]
		public bool Groups { get; set; }


		[DataMember(Name="include_segment_file_sizes")]
		public bool IncludeSegmentFileSizes { get; set; }


		[DataMember(Name="level")]
		public Level Level { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="types")]
		public List<string> Types { get; set; }

	}
}
